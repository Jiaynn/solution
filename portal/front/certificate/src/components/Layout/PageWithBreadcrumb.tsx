/*
 * @file component PageWithBreadcrumb
 * @author zhuhao <zhuhao@qiniu.com>
 */

import React from 'react'
import Page from 'portal-base/common/components/Page'

import Breadcrumb from '../Breadcrumb'

export interface IPageWithBreadcrumb {
  sideItems?: React.ReactNode
}

export default function PageWithBreadcrumb({ children, ...restProps }: React.PropsWithChildren<IPageWithBreadcrumb>) {
  return (
    <Page className="comp-page-wrapper" header={<Breadcrumb {...restProps} />}>
      <div className="comp-page-content">
        {children}
      </div>
    </Page>
  )
}
