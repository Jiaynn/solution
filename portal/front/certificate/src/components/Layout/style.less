/*
 * @file style for Layout
 * @author nighca <nighca@live.cn>
 */

@import "~portal-base/common/utils/style/index.less";

.main-popup-container {
  display: flex;
  min-height: 100%;
  flex-direction: column;

  & > div {
    flex: auto;
  }
}

.comp-page-wrapper {
  &.has-footer {
    padding-bottom: 76px;
  }

  .page-header-alert {
    margin-bottom: 20px;
  }

  .comp-page-content {
    overflow: auto;
  }

  .comp-page-footer {
    height: 76px;
    position: fixed;
    z-index: 101;
    left: 0;
    // 距离右侧滚动条的间距
    right: 15px;
    bottom: 0;
    background-color: @icecream-color-grey-1;
  }
}
