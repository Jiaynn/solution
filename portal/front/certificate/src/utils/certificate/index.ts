/*
 * @file component certificate utils
 * @author Yao Jingtian <yncst233@gmail.com>
 */

import { sortBy, last, without, uniq, values } from 'lodash'

import { OrderType } from '../../apis/ssl'
import {
  sslMap, canBeClosedStatuses, SSLDomainType, CertType,
  OrderStatus, ProductShortName, SslBrand, isDVSSLType
} from '../../constants/ssl'
import { isOEM } from '../../constants/env'
import { isStatusDone, isStatusPending } from './status'

interface ICertBrandInfo {
  brand: SslBrand
  certType: CertType
}

// 根据 productShortName 获取证书品牌、类型
export function shortNameToInfo(productShortName?: ProductShortName): ICertBrandInfo {
  let brand: SslBrand
  let certType: CertType
  Object.keys(sslMap).some((sslBrand: SslBrand) => Object.keys(sslMap[sslBrand]).some((sslType: CertType) => {
    if (sslMap[sslBrand][sslType]!.shortName === productShortName) {
      brand = sslBrand
      certType = sslType
      return true
    }
    if (productShortName && sslMap[sslBrand][sslType]!.domainShortNameMap) {
      const names = values(sslMap[sslBrand][sslType]!.domainShortNameMap)
      if (names.indexOf(productShortName) > -1) {
        brand = sslBrand
        certType = sslType
        return true
      }
    }
    return false
  }))
  return {
    brand: brand!,
    certType: certType!
  }
}

export function getLatestDnsNames(replaceOrders: any[], dnsNames?: string[]): string[] {
  if (!replaceOrders || replaceOrders.length === 0) {
    return dnsNames || []
  }
  const list = sortBy(replaceOrders, 'create_time')
    .filter(order => order.state === 1)
  if (list.length > 0) {
    return last(list).dnsnames.split(',')
  }
  return dnsNames || []
}

export function hasRepeatItem(arr: any[], compareArr?: any[]) {
  if (arr && compareArr) {
    return without(arr.slice(), ...compareArr.slice()).length !== arr.length
  }
  return uniq(arr.slice()).length !== arr.length
}

/**
 * 从后端接口字段 renewable 来判断
 * 历史前端逻辑：已签发/已重颁发 过期前 30 天的非重颁发订单，显示【续费】(若 notAfter <= 0，则根据 providerRenewable 字段)
 */
export function canRenew(renewable: boolean) {
  return isOEM ? false : renewable
}

interface ICanAddDomainParams {
  notAfter: number,
  state: number,
  productType: SSLDomainType,
  orderType: OrderType
}

// 已签发/已重颁发 非重颁发 未过期的多域名、多域名泛域名订单，显示【添加域名】
export function canAddDomain(params: ICanAddDomainParams) {
  const { notAfter, state, productType, orderType } = params
  const now = Date.now()
  if (!state || !isStatusDone(state)) {
    return false
  }
  if ([SSLDomainType.Multiple, SSLDomainType.MultipleWildcard].indexOf(productType) === -1) {
    return false
  }
  if (orderType === OrderType.Replace) {
    return false
  }
  if (!notAfter) {
    return false
  }
  return (
    notAfter <= 0
    || now < notAfter * 1000
  )
}

interface ICanUploadConfirmationParams {
  state: number,
  productShortName: ProductShortName,
  uploadConfirmLetter: boolean
}

// 非 DV 且待确认状态才显示【上传确认函】
export function canUploadConfirmation(params: ICanUploadConfirmationParams) {
  const { state, productShortName, uploadConfirmLetter } = params
  if (!state || !isStatusPending(state)) {
    return false
  }

  const sslInfo = shortNameToInfo(productShortName)

  return !uploadConfirmLetter && !isDVSSLType(sslInfo.certType)
}

export function canClose(state: number): boolean {
  return canBeClosedStatuses.indexOf(state) !== -1
}

// 过期 7 天前的证书都能够部署
export function canCertDeploy(notAfter: number): boolean {
  return !isOEM && ((notAfter - 7 * 24 * 3600) > (Date.now() / 1000))
}

// 判断证书类型是否为私有证书
export function isPrivateCert(productShortName: ProductShortName) {
  return !productShortName
}

// 已经完成的订单，并且满足证书部署条件的订单能够部署
export function canOrderDeploy(notAfter: number, state: OrderStatus): boolean {
  return isStatusDone(state) && canCertDeploy(notAfter)
}

// 2 年的续期证书需要补充文案
export function getCertYearTipsByYear(year: number): string {
  return year === 2 ? '续期证书' : ''
}

// 自动续期的证书、订单需要补充文案
export function getCertYearTipsByAutoRenew(autoRenew: boolean): string {
  return autoRenew ? '续期证书' : ''
}
