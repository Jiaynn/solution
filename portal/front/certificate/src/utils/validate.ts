/*
 * @file formstate validate
 * @author zhu hao <zhuhao@qiniu.com>
 */

import { standardDomainRegx, wildcardDomainRegx } from '../constants/domain'

export function validate(criteria: boolean, invalidMsg: string) {
  return criteria ? invalidMsg : null
}

export function notEmpty(value: unknown, name = '') {
  return validate(value == null || !value, `${name}不可以为空`)
}

export function inputValid(value: string) {
  return validate(/[`~～!！@#$%^&*_+<>?:"“”{},，。./;；'‘’[\]]/.test(value), '不得含有非法字符')
}

export function mustContainChinese(value: string, name = '') {
  return validate(!/[\u4e00-\u9fa5]/.test(value), `${name}应包含中文字符`)
}

export function postCodeValid(value: string) {
  return validate(!/^\d{5,6}(?!\d)$/.test(value), '邮编格式不正确')
}

export function areaCodeValid(value: string) {
  return validate(!/^((0\d{2,3}))$/.test(value), '区号格式不正确')
}

export function telephoneValid(value: string) {
  return validate(!/^(\d{7,8})(-(\d{3,}))?$/.test(value), '座机号格式不正确')
}

export function mobilephoneValid(value: string) {
  return validate(!/^\d{11}$/.test(value), '手机号格式不正确')
}

export function emailValid(value: string) {
  return validate(!/^\w+([-+.]\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$/.test(value), '邮箱格式不正确')
}

export function wildcardDomainValid(value: string) {
  return validate(!wildcardDomainRegx.test(value), '泛域名格式不正确')
}

export function standardDomainValid(value: string) {
  return validate(!standardDomainRegx.test(value), '标准域名格式不正确')
}

export function onlySupportChineseAndCharacter(value: string) {
  return validate(!/^[a-zA-Z0-9_\s\-*.()\u4e00-\u9fa5]+$/.test(value), '仅支持中英文字符、数字、空格、_、-、*、.、()')
}
