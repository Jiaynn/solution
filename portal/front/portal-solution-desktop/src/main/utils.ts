import { join } from 'path'
import { BrowserWindow, dialog, session, shell } from 'electron'
import cmd from 'node-cmd'

import { pageUrl } from './config'
import { ElectronBridgeApi } from '../preload'

import icon from '../../resources/logo.png?asset'

export const createWindow = (): BrowserWindow => {
  // Create the browser window.
  const minWidth = 1024
  const minHeight = 768
  const mainWindow = new BrowserWindow({
    width: minWidth,
    height: minHeight,
    minWidth,
    minHeight,
    show: false,
    autoHideMenuBar: true,
    titleBarStyle: 'hidden',
    ...(process.platform === 'linux' ? { icon } : {}),
    webPreferences: {
      preload: join(__dirname, '../preload/index.js'),
      sandbox: false,
      nodeIntegration: true
    }
  })

  mainWindow.center()

  mainWindow.on('ready-to-show', () => {
    mainWindow.show()
  })

  mainWindow.webContents.setWindowOpenHandler((details) => {
    shell.openExternal(details.url)
    return { action: 'deny' }
  })

  // HMR for renderer base on electron-vite cli.
  // Load the remote URL for development or the local html file for production.
  mainWindow.loadURL(pageUrl)

  return mainWindow
}

export const persistentCookie = (): void => {
  session.defaultSession.cookies.on('changed', (_, cookie) => {
    if (cookie.session && cookie.name === 'PORTAL_SESSION') {
      session.defaultSession.cookies.set({
        url: pageUrl,
        name: cookie.name,
        value: cookie.value,
        domain: cookie.domain,
        path: cookie.path,
        secure: cookie.secure,
        httpOnly: cookie.httpOnly,
        expirationDate: Math.floor(Date.now() / 1000) + 60 * 60 * 24 * 2,
        sameSite: cookie.sameSite
      })
    }
  })
}

type Platform = Parameters<ElectronBridgeApi['openEditor']>[0]['platform']
export const callEditor = (platform: Platform, dest: string): void => {
  if (process.platform === 'darwin') {
    // 运行在 macOS 上
    if (platform === 'Android') {
      cmd.run(`open -a /Applications/Android\\ Studio.app ${dest}`)
      return
    }
    if (platform === 'iOS') {
      cmd.run(`open -a /Applications/Xcode.app ${dest}`)
      return
    }
  }
  if (process.platform === 'win32') {
    // 运行在 Windows 上
    // ...
    return
  }
  if (process.platform === 'linux') {
    // 运行在 Linux 上
    // ...
  }
  dialog.showMessageBox({ message: '打开失败，请检查编辑器是否安装或项目是否存在' })
}
