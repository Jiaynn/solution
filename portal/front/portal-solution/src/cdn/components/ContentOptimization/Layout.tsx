/**
 * @desc 内容优化 layout
 * @author yaojingtian <yaojingtian@qiniu.com>
 */

import React, { PropsWithChildren } from 'react'
import { useInjection } from 'qn-fe-core/di'
import Page from 'portal-base/common/components/Page'

import LayoutWithTabs from 'cdn/components/Layout/WithTabs'
import Routes from 'cdn/constants/routes'

export default function ContentOptimizationLayout(props: PropsWithChildren<{}>) {
  const routes = useInjection(Routes)

  const tabs = [{
    path: routes.optimizationVideo,
    name: '视频瘦身',
    featureConfigKey: 'FUSION.FUSION_VIDEO_SLIM'
  }]

  return (
    <Page className="comp-page-content-optimization" header={null} hasSpace={false}>
      <LayoutWithTabs tabs={tabs}>{props.children}</LayoutWithTabs>
    </Page>
  )
}
