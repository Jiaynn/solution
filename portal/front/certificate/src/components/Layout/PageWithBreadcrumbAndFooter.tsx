/*
 * @file component PageWithBreadcrumbAndFooter
 * @author zhuhao <zhuhao@qiniu.com>
 */

import React from 'react'
import Page from 'portal-base/common/components/Page'

import Breadcrumb from '../Breadcrumb'
import { IPageWithBreadcrumb } from './PageWithBreadcrumb'

export interface IPageWithBreadcrumbAndFooter extends IPageWithBreadcrumb {
  footer: React.ReactNode
}

export default function PageWithBreadcrumbAndFooter(props: React.PropsWithChildren<IPageWithBreadcrumbAndFooter>) {
  return (
    <Page className="comp-page-wrapper has-footer" header={<Breadcrumb {...props} />}>
      <div className="comp-page-content">
        {props.children}
      </div>
      <div className="comp-page-footer">
        {props.footer}
      </div>
    </Page>
  )
}
