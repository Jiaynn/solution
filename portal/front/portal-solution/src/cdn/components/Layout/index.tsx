/**
 * @file component Layout
 * @author nighca <nighca@live.cn>
 */

import React, { ReactNode } from 'react'
import { observer } from 'mobx-react'
import { useInjection } from 'qn-fe-core/di'
import { UserInfoStore } from 'portal-base/user/account'
import QiniuLayout from 'portal-base/common/components/Layout'

import Loading from './Loading'
import SubSidebar from './SubSidebar'
import Content from './Content'

import './style.less'

export interface Props {
  children: ReactNode
}

export default observer(function Layout(props: Props) {
  const userInfoStore = useInjection(UserInfoStore)

  const loading = (!userInfoStore.isLoaded || userInfoStore.isGuest)
    ? <Loading />
    : null

  const main = (
    <QiniuLayout>
      <Content sidebar={<SubSidebar />}>{props.children}</Content>
    </QiniuLayout>
  )

  return (
    <>
      {loading}
      {main}
    </>
  )
})
