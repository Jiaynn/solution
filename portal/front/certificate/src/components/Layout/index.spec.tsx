/**
 * @file test cases for component Layout
 * @author [Yao Jingtian] [yncst233@gmail.com]
 */
import React from 'react'

import { RendererUtils as Renderer } from '../../utils/test'

import Layout from '.'

it('renders correctly', () => {
  const renderer = new Renderer()
  const tree = renderer.create(
    <Layout />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
