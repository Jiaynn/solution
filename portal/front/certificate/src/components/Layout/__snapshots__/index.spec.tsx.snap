// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=""
>
  <div
    className="icecream-toaster-container comp-toaster"
  />
  <div
    className="static-notification"
  />
  <div
    className="comp-document-popup react-draggable"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    style={
      Object {
        "boxSizing": "border-box",
        "cursor": "move",
        "display": "none",
        "flexShrink": 0,
        "height": "470px",
        "left": 0,
        "maxHeight": 9007199254740991,
        "maxWidth": 9007199254740991,
        "minHeight": 235,
        "minWidth": 360,
        "position": "absolute",
        "top": 0,
        "transform": "translate(0px,0px)",
        "userSelect": "auto",
        "width": "360px",
      }
    }
    transform={null}
  >
    <div
      className="comp-document-popup-header"
    >
      <svg
        className="ctl-icon disabled"
        height="1em"
        onClick={[Function]}
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8.39895924,0.398959236 L9.60104076,1.60104076 L4.20191847,6.99995924 L16,7 L16,9 L4.20191847,8.99995924 L9.60104076,14.3989592 L8.39895924,15.6010408 L0.797918472,8 L8.39895924,0.398959236 Z"
          fill="currentColor"
        />
      </svg>
      <strong>
        帮助文档
      </strong>
      <svg
        className="ctl-icon"
        height="1em"
        onClick={[Function]}
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M12.2928932,2.29289322 L13.7071068,3.70710678 L9.41389322,7.99989322 L13.7071068,12.2928932 L12.2928932,13.7071068 L7.99989322,9.41389322 L3.70710678,13.7071068 L2.29289322,12.2928932 L6.58589322,7.99989322 L2.29289322,3.70710678 L3.70710678,2.29289322 L7.99989322,6.58589322 L12.2928932,2.29289322 Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <iframe
      className="comp-document-popup-body"
      src={null}
    />
    <div
      className="comp-document-popup-footer"
    />
    <div>
      <div
        className=""
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "cursor": "row-resize",
            "height": "10px",
            "left": "0px",
            "position": "absolute",
            "top": "-5px",
            "userSelect": "none",
            "width": "100%",
          }
        }
      />
      <div
        className=""
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "cursor": "col-resize",
            "height": "100%",
            "position": "absolute",
            "right": "-5px",
            "top": "0px",
            "userSelect": "none",
            "width": "10px",
          }
        }
      />
      <div
        className=""
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "bottom": "-5px",
            "cursor": "row-resize",
            "height": "10px",
            "left": "0px",
            "position": "absolute",
            "userSelect": "none",
            "width": "100%",
          }
        }
      />
      <div
        className=""
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "cursor": "col-resize",
            "height": "100%",
            "left": "-5px",
            "position": "absolute",
            "top": "0px",
            "userSelect": "none",
            "width": "10px",
          }
        }
      />
      <div
        className=""
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "cursor": "ne-resize",
            "height": "20px",
            "position": "absolute",
            "right": "-10px",
            "top": "-10px",
            "userSelect": "none",
            "width": "20px",
          }
        }
      />
      <div
        className=""
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "bottom": "-10px",
            "cursor": "se-resize",
            "height": "20px",
            "position": "absolute",
            "right": "-10px",
            "userSelect": "none",
            "width": "20px",
          }
        }
      />
      <div
        className=""
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "bottom": "-10px",
            "cursor": "sw-resize",
            "height": "20px",
            "left": "-10px",
            "position": "absolute",
            "userSelect": "none",
            "width": "20px",
          }
        }
      />
      <div
        className=""
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "cursor": "nw-resize",
            "height": "20px",
            "left": "-10px",
            "position": "absolute",
            "top": "-10px",
            "userSelect": "none",
            "width": "20px",
          }
        }
      />
    </div>
  </div>
  <div
    className="comp-global-loading"
  >
    <div
      className="global-loading-content"
    >
      <img
        alt="loading"
        src="https://qiniu.staticfile.org/loading.gif"
      />
      平步云端，数据为先
    </div>
  </div>
</div>
`;
